src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v2/NetworkControllerV2.java

```diff
@@ -0,0 +1,67 @@
+/* Copyright 2023 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.northbound.controller.v2;
+
+import org.openkilda.messaging.error.ErrorType;
+import org.openkilda.messaging.error.MessageException;
+import org.openkilda.messaging.payload.network.PathValidationDto;
+import org.openkilda.northbound.dto.v2.flows.PathValidateResponse;
+import org.openkilda.northbound.service.NetworkService;
+
+import io.swagger.annotations.ApiOperation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.concurrent.CompletableFuture;
+
+@RestController
+@RequestMapping("/v2/network")
+public class NetworkControllerV2 {
+
+    private final NetworkService networkService;
+
+    @Autowired
+    public NetworkControllerV2(NetworkService networkService) {
+        this.networkService = networkService;
+    }
+
+    /**
+     * Validates that a given path complies with the chosen strategy and the network availability.
+     * It is required that the input contains path nodes. Other parameters are optional.
+     * @param pathValidationDto a payload with a path and additional flow parameters provided by a user
+     * @return either a successful response or the list of errors
+     */
+    @GetMapping(path = "/path/check")
+    @ApiOperation(value = "Validates that a given path complies with the chosen strategy and the network availability")
+    @ResponseStatus(HttpStatus.OK)
+    public CompletableFuture<PathValidateResponse> validateCustomFlowPath(
+            @RequestBody PathValidationDto pathValidationDto) {
+
+        if (pathValidationDto == null
```

Comment:
Shouldn't this stuff to be in some Validator class?

Reply:
I think it is quite common to sanitize the user's input or put some defaults in a controller before passing to a service. Like they do a bean validation in hibernate right in the method signature:
```
@GetMapping(path = "/someUrl")
public ResponseEntity<List<Object>> method(@RequestParam(value = "list", required = false) List<@Size(min = 1, max = 2) String> list)
```
Is it worth it to create a separate class for these 3 conditions?

Reply:
I believe we should stay unified in all our decisions. If we leave validation to the separate classes in other parts, we should create a new one here. 
If we allow simple validations in other controllers, we may leave the one here as well. Please, check the other code.

Reply:
There are some examples of validations or some simple data transformation in other controllers. I'd leave it like that. It is really just an input sanity check: provide at least two points in a path is mandatory. Just a more verbose version of `required = true`